src/analysis/souffle/sensitivity_analysis.dl

```diff
@@ -0,0 +1,95 @@
+#ifndef SENSITIVITY_ANALYSIS_DL_
+#define SENSITIVITY_ANALYSIS_DL_
+//this file defines the calculations for sensitivity analysis 
+//to be used in dp calculations
+
+#include "src/analysis/souffle/taint.dl"
+
+//one isSqlOperator fact per each sql operator, currently supporting one output
+//referring to results as 'value' (sqlValue declared below) as per the discussion in the doc 
+.decl isSqlOperator(sqlOperator: Operation)
+isSqlOperator([owner, operator, result, input_list, attribute_list]) :- isOperation([owner, operator, result, input_list, attribute_list]), match("sql.*", operator). 
+
+
+.decl isSqlValue(value: AccessPath)
+isSqlValue(value) :- isSqlOperator([_, _, value, _, _]).
+
+//keeping track of all initial inputs as they all have sensitivity score 1
+.decl isInitialInput (ingress_point: AccessPath)
+
+//helper functions to flatten input lists
+.decl intermediateFlatten(list:OperandList, sublist:OperandList, list_element:AccessPath)
+
+intermediateFlatten(list, sublist, list_element) :- isSqlOperator([_, _, _, list, _]), list = [list_element, sublist].
+intermediateFlatten(list, sublist, list_element) :- intermediateFlatten(list, [list_element, sublist], _).
+
+.decl flatten(list:OperandList, list_element:AccessPath)
+flatten(list, list_element) :- intermediateFlatten(list, _, list_element).
+
+//flattening each individual input per value
+.decl isSqlInput(value: AccessPath, sql_input: AccessPath)
+
+isSqlInput(value, sql_input) :- isSqlOperator([_, _, value, input_list, _]), 
+                              flatten(input_list, sql_input). 
+
+//keeping track of sensitivity score per each input at index 
+//used to calculate the maximum input sensitivity 
+.decl sqlValueHasSensitivityScoreAtIndex(value: AccessPath, sensitivity_score: number, index: number) 
```

Comment:
This is capturing max of sensitivity for an `operation` (not value) upto an index, right? Can we name it and change the signature accordingly? e.g., 

`sqlOperationHasSensitivityScoreUptoInputIndex(op: Operation, sensitivity_score: number, index: number)`

You can use any other name that you think is meaningful.  The key thing is that we should key on `Operation` instead of `AccessPath`?

Reply:
We're currently tracking sensitivities per value though (value being the output of an sqlOperation). I implemented this based on our discussions in the design doc. I am not sure I see an advantage to converting to Operation (as we would have to add checks for Operation's output everywhere). What do you think?

Reply:
Changed.